lib/modem_info/modem_info.c

```diff
@@ -720,6 +722,116 @@ int modem_info_rsrp_register(rsrp_cb_t cb)
 	return 0;
 }
 
+/* FW UUID is 36 characters: XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX */
+#define FW_UUID_SIZE 37
+
+int modem_info_get_fw_uuid(char *buf, size_t buf_size)
+{
+	int ret;
+
+	if (buf_size < FW_UUID_SIZE) {
+		return -EINVAL;
+	}
+
+	ret = nrf_modem_at_scanf("AT%XMODEMUUID",
+				 "%%XMODEMUUID: %" STRINGIFY(FW_UUID_SIZE) "[^\r\n]",
+				 buf);
+	if (ret != 1) {
+		LOG_ERR("Could not get FW ID, error: %d", ret);
+		return -EIO;
```

Comment:
It would be nice to return a more specific error, but `nrf_modem_at_scanf` returns NRF-error codes that should not be exposed from this library. According to `nrf_modem_at.h`, `nrf_modem_at_scanf` can produce 4 different errors:

```
 * @retval -NRF_EPERM The Modem library is not initialized.
 * @retval -NRF_EFAULT @c cmd or @c fmt are @c NULL.
 * @retval -NRF_EBADMSG No arguments were matched.
 * @retval -NRF_ENOMEM Not enough shared memory for this request.
 ```

 [Zephyr have the almost the same error defines](https://docs.zephyrproject.org/apidoc/latest/group__system__errno.html#ga251e9b536ed96ccb16aa28ca2d5bd656), only without the `NRF_` prefix. So mapping to Zephyr error codes should be trivial.